R/tools.R

```diff
@@ -35,34 +42,89 @@ bowtie2 <- function(tool_args = "", inputs = NULL, output_path = NULL, database_
 #' @examples
 #' \dontrun{
 #' cutadapt(tool_args = "-a AATTCCGG")
-#' cutadapt(input_path = "C://Users/YourName/Documents/my_input_file.fastq", tool_args = "-a AATTCCGG")
+#' cutadapt(inputs = "C://Users/YourName/Documents/my_input_file.fastq", tool_args = "-a AATTCCGG")
 #' }
 #'
 #' @export
 cutadapt <- function(tool_args, inputs = NULL, output_path = NULL) {
   inputs <- .validate.inpath(inputs)
   output_path <- .validate.outpath(output_path)
-  toolchest_client$cutadapt(inputs, output_path, tool_args)
+  toolchest_args <- list(
+    inputs = inputs,
+    output_path = output_path,
+    tool_args = tool_args
+  )
+  .do.toolchest.call(toolchest_client$cutadapt, toolchest_args)
 }
 
 #' Kraken 2 Client
 #'
 #' Starts a query for Kraken 2 using Toolchest.
 #'
-#' (Currently, only single .fastq inputs are supported.)
-#'
 #' If left unspecified, inputs and output_path can be selected by the user
 #' manually.
 #'
 #' @param tool_args (optional) Additional arguments to be passed to Kraken 2.
 #' @param inputs Path or list of paths (client-side) to be passed in as input.
 #' @param output_path Path (client-side) where the output will be downloaded.
+#' @param database_name (optional) Name (string) of database to use for Kraken 2 alignment.
+#' @param database_version (optional) Version (string) of database to use for Kraken 2 alignment.
+#'
+#' @note If the database is unspecified, Kraken 2 will use the standard
+#' database, using index files generated by Ben Langmead's lab at
+#' \url{https://benlangmead.github.io/aws-indexes/k2}.
 #'
 #' @export
-kraken2 <- function(tool_args = "", inputs = NULL, output_path = NULL) {
-  inputs <- .validate.inpath(inputs)
+kraken2 <- function(tool_args = "", inputs = NULL, output_path = NULL,
+                    database_name = NULL, database_version = NULL) {
+  inputs <- .validate.inpath(inputs, choose_multiple = TRUE)
```

Comment:
Did you test Kraken with multiple inputs?

Reply:
Kraken with multiple inputs is currently disabled on the Python side. We'll eventually want to support this, so I enabled this option, but we can remove it for now and re-add it once it's good to go on Python.

Reply:
Sounds good

Reply:
(meaning we should probably disable until enabled on the Python side)